src/main/java/me/jellysquid/mods/phosphor/mixin/chunk/MixinWorldChunk.java

```diff
@@ -37,28 +40,9 @@
         final int startX = this.pos.getStartX();
         final int startZ = this.pos.getStartZ();
 
-        List<BlockPos> list = new ArrayList<>();
-        ChunkSection[] chunkSections = this.sections;
-
-        for (ChunkSection section : chunkSections) {
-            if (section == null || section.isEmpty()) {
-                continue;
-            }
-
+        return Arrays.stream(this.sections).flatMap(section -> {
             final int startY = section.getYOffset();
-
```

Comment:
I know that the Stream API is not necessarily the most performant, however the return type of the method is a stream so I have replaced the use of an array, which is then streamed with a direct stream implementation.

Reply:
The point of this code was to replace the stream implementation that vanilla minecraft has. Comparing your code to the vanilla implementation, I cannot see where it could be faster. Additionally your code looks like it will allocate more than vanilla does. (Lithium also seems to allocate more than vanilla, vanilla uses a single BlockPos Mutable

Reply:
I benchmarked the whole time. My stream implementation is faster than the vanilla implementation, but the ArrayList method is currently even faster. That's why I rolled back this commit. I hope that I have met all the criteria for this pull request to be accepted.